cli/tests/pcluster/validators/test_cluster_validators.py

```diff
@@ -1790,6 +1792,167 @@ def test_new_storage_multiple_subnets_validator(queues, new_storage_count, failu
+            "latency and costs.",
+        ),
+        (
+            [
+                SlurmQueue(
+                    name="multi-az-queue-match",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-1", "subnet-2"],
+                    ),
+                ),
+                SlurmQueue(
+                    name="multi-az-queue-match",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-4", "subnet-5"],
+                    ),
+                ),
+            ],
+            [
+                {"subnet-1": "us-east-1a", "subnet-2": "us-east-1b"},
+                {"subnet-4": "us-east-1a", "subnet-5": "us-east-1b"},
+            ],
+            ["us-east-1a", "us-east-1b"],
+            None,
+            "",
+        ),
+        (
+            [
+                SlurmQueue(
+                    name="multi-az-queue-mismatch",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-2", "subnet-3"],
+                    ),
+                ),
+            ],
+            [{"subnet-2": "us-east-1b", "subnet-3": "us-east-1c"}],
+            ["us-east-1a", "us-east-1b"],
+            FailureLevel.INFO,
+            "Your configuration for Queue multi-az-queue-mismatch includes multiple subnets and external "
+            "shared storage configuration. Accessing a shared storage from different AZs can lead to increased "
+            "latency and costs.",
+        ),
+        (
+            [
+                SlurmQueue(
+                    name="multi-az-queue-partial-match",
+                    compute_resources=[],
+                    networking=SchedulerPluginQueueNetworking(
+                        subnet_ids=["subnet-1", "subnet-2"],
+                    ),
+                ),
+            ],
+            [{"subnet-1": "us-east-1a", "subnet-2": "us-east-1b"}],
+            ["us-east-1b"],
+            FailureLevel.INFO,
+            "Your configuration for Queue multi-az-queue-partial-match includes multiple subnets and external "
+            "shared storage configuration. Accessing a shared storage from different AZs can lead to increased "
+            "latency and costs.",
+        ),
+        (
+            [
+                SlurmQueue(
+                    name="multi-az-queue-match",
+                    compute_resources=[],
+                    networking=SchedulerPluginQueueNetworking(
+                        subnet_ids=["subnet-1", "subnet-2"],
+                    ),
+                ),
+            ],
+            [{"subnet-1": "us-east-1a", "subnet-2": "us-east-1b"}],
+            ["us-east-1a", "us-east-1b"],
+            None,
+            "",
+        ),
```

Comment:
missing test case with more than 1 queue where none of the queue is in the same AZ of the FS